crates/kitsune_p2p/fetch/src/lib.rs

```diff
@@ -25,25 +27,27 @@ pub use respond::*;
 #[serde(tag = "type", rename_all = "camelCase")]
 pub enum FetchKey {
     /// Fetch via region.
-    Region {
-        /// The region coordinates to fetch
-        region_coords: RegionCoords,
-    },
+    Region(RegionCoords),
 
     /// Fetch via op hash.
-    Op {
-        /// The hash of the op to fetch.
-        op_hash: KOpHash,
-    },
+    Op(KOpHash),
 }
 
 /// A fetch "unit" that can be de-duplicated.
-#[derive(Debug, Clone, PartialEq, serde::Deserialize, serde::Serialize)]
-#[serde(tag = "type", rename_all = "camelCase")]
-pub struct FetchRequest {
+#[derive(Debug, Clone, PartialEq)]
+pub struct FetchQueuePush {
     /// Description of what to fetch.
     pub key: FetchKey,
 
+    /// The space this op belongs to
+    pub space: KSpace,
+
+    /// The source to fetch the op from
+    pub source: FetchSource,
+
+    /// The approximate size of the item
+    pub size: Option<RoughInt>,
+
```

Comment:
I just moved space and source into this struct (they were supplied independently before), but `size` is the new addition here to the push parameters.